designer/client/form-details.js

```diff
@@ -45,6 +50,23 @@ class FormDetails extends React.Component {
     }
   };
 
+  validate = () => {
+    let validationErrors = false;
+    const { data } = this.props;
+    const { title, feedbackForm, selectedFeedbackForm } = this.state;
+
+    let titleHasErrors = !title || title.trim().length < 1;
+    this.setState((prevState, props) => ({
+      errors: {
+        ...prevState.errors,
+        title: titleHasErrors,
+      },
+    }));
+
+    validationErrors = titleHasErrors;
+    return validationErrors;
+  };
```

Comment:
Careful because`setState` is asynchronous and destructing the state inside onSubmit might give a false positive. 
IMO `validate` should be just a predicate without any side-effects (no state changes), because it could be used in other situations which doesn't necessarily require a state change, then whoever calls validate is responsible for managing the state changes.

Reply:
```javascript
onSubmit() {
 const errors = this.validate()
 
 if (errors) {
   return thit.setState({ errors })
 }

  this.setState({ isSaving: true });
  // etc
}

```

Reply:
at the moment I am using validate with side effects - it helps with 2 problems
1) many of the sub components which retain state and associated validation logic - are validated  via refs(e.g fee edit and fee item, component-type-edit.js) ,  so to keep it consistent I followed the same pattern even in components which don't have subcomponents like form-details
2) if we decide to use validation on onBlur instead of onSubmit event this method can be refactored to only validate and set state of the field which is passed as argument